src/runtime/virtcontainers/kata_agent.go

```diff
@@ -488,9 +506,16 @@ func (k *kataAgent) setupSharedPath(ctx context.Context, sandbox *Sandbox) error
 	if err := bindMount(ctx, mountPath, sharePath, true, "slave"); err != nil {
 		return err
 	}
+	defer func() {
+		if err != nil {
+			if umountErr := syscall.Unmount(sharePath, syscall.MNT_DETACH|UmountNoFollow); umountErr != nil {
+				k.Logger().WithError(umountErr).Errorf("failed to unmount vm share path %s", sharePath)
+			}
+		}
+	}()
 
 	// Setup sandbox bindmounts, if specified:
-	if err := setupSandboxBindMounts(sandbox); err != nil {
+	if err := k.setupSandboxBindMounts(sandbox); err != nil {
```

Comment:
Since you have make return value as named `err error`, here `err := k.setupSandboxBindMounts(sandbox)` should be replaced by `err = k.setupSandboxBindMounts(sandbox)` by removing the `:`, otherwise line 510 may get an `nil` value.

And these suggests are not error, but recommend to change:

- other `err := ` -> `err =` in this function
- use `return` replace `return err`